pkgs/by-name/fl/flashprog/package.nix

```
@@ -24,19 +26,29 @@ stdenv.mkDerivation rec {
     installShellFiles
     pkg-config
   ];
-
   buildInputs = [
     libftdi1
-    libjaylink
     libusb1
   ] ++ lib.optionals (!stdenv.isDarwin) [
-    libgpiod_1
     pciutils
+  ] ++ lib.optionals with_jlink [
+    libjaylink
+  ] ++ lib.optionals (stdenv.isLinux && with_gpio) [
```

```suggestion
  ] ++ lib.optionals (with_gpio) [
```
having the option true and still not adding the binary is not great

---

libgpio is specific to Linux. The isLinux condition needs to stay.

---

This is not the typical code on Nixpkgs.
We usually employ

```nix
{
... 
, enableXYZ? systemHasABC
... 
}:

{
... 
  confstring = if enableXYZ then "--enable-xyz" else "disable-xyz";
... 
} 
```

in such cases.
The other format

```nix
{
... 
, enableXYZ? true
... 
}:

{
... 
  confstring = if (enableXYZ && systemHasABC) then "--enable-xyz" else "disable-xyz";
... 
} 
```

makes `enableXYZ` a no-op when `systemHasABC` is false, violating the intention of the user that sets `enableXYZ`.

If it is so essential and imperative to build with `enableXYZ` only when `systemHasABC` is true, then use an assertion:

```nix
{
... 
, enableXYZ? systemHasABC
... 
}:

assert enableXYZ - > systemHasABC;
{
... 
  confstring = if enableXYZ then "--enable-xyz" else "disable-xyz";
... 
} 
```

---

I removed the isLinux condition here and set withGpio defaults to true for isLinux and to false otherwise. This way darwin users get an explictit error when setting withGpio to true instead of just ignoring the option. I think this is more transparent to the user.